### PR TITLE
Move `OwnedById` and `UpdatedById` to `type_ids` table

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

For future work and to avoid duplication we can move these types to the shared `type_ids` table

## 🔗 Related links

- This is required for [this Asana task](https://app.asana.com/0/0/1203774687353268/f) _(internal)_

## 🔍 What does this change?

- Change the database layout
- Adjust inserting logic
- Adjust querying logic
- Drive-by: Query base URI and the version from the `type_ids` table as well

### CODE REVIEW

File: packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/read.rs [hunk 1]
```diff
@@ -56,23 +63,26 @@ where
...
                     .into_report()
                     .change_context(QueryError)?;
-                let owned_by_id = OwnedById::new(row.get(owned_by_id_index));
-                let updated_by_id = UpdatedById::new(row.get(updated_by_id_path_index));
 
                 Ok(T::new(
                     record,
                     OntologyElementMetadata::new(
-                        OntologyTypeEditionId::from(&versioned_uri),
+                        OntologyTypeEditionId::new(base_uri, version),
```
Comment (thehabbos007) [inline packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/read.rs:85]: The change made here to change the queried`versioned_uri` seems unrelated to the rest of the changes. Was this done to prevent looking into `schema->>$id` to get the versioned URI?

File: packages/graph/hash_graph/lib/graph/src/store/postgres/mod.rs [hunk 2]
```diff
@@ -345,11 +353,11 @@ where
 
         let version_id = VersionId::new(Uuid::new_v4());
         self.insert_version_id(version_id).await?;
-        self.insert_uri(&uri, version_id).await?;
-
-        self.insert_with_id(version_id, database_type, owned_by_id, updated_by_id)
+        self.insert_uri(&uri, version_id, owned_by_id, updated_by_id)
             .await?;
 
+        self.insert_with_id(version_id, database_type).await?;
```
Comment (thehabbos007) [inline packages/graph/hash_graph/lib/graph/src/store/postgres/mod.rs:359]: I suppose the SQL `function`s are yet to be made, are they going to be in a follow up PR?

**Review by thehabbos007 — COMMENTED** [review]
Looks good to me! I have a couple questions, but other than that I'm happy to merge

File: packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/read.rs (hunk 1 above)
Comment (TimDiekmann) [reply packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/read.rs:85]: Yes, I wanted to avoid the schema lookup here. We have to look in the `type_ids` table anyway, so there is no reason to not directly look them up there.

File: packages/graph/hash_graph/lib/graph/src/store/postgres/mod.rs (hunk 2 above)
Comment (TimDiekmann) [reply packages/graph/hash_graph/lib/graph/src/store/postgres/mod.rs:359]: Yes, I'm currently cleaning up my branch and will open a PR with the functions in a couple minutes 🚀

**Review by thehabbos007 — APPROVED** [review]
